plugins/qlcplus_4.12.1/import.js

```diff
@@ -601,6 +620,222 @@ const qlcPlusCapabilityPresets = {
+      delete cap.slotNumber;
+    }
+
+    return cap;
+  },
+  Effect: ({ capabilityName }) => {
+    const cap = {
+      type: `Effect`,
+      effectName: `` // set it first here so effectName is before speedStart/speedEnd
+    };
+    cap.effectName = getSpeedGuessedComment(capabilityName, cap);
+
+    if (/\bsound\b/i.test(cap.effectName)) {
+      cap.soundControlled = true;
+    }
+
+    return cap;
+  },
+  Maintenance: ({ capabilityName }) => ({
+    type: `Maintenance`,
+    comment: capabilityName
+  }),
+  Pan: ({ channelName, capabilityName, panMax }) => {
+    const cap = {};
+
+    if (channelName.match(/continuous/i)) {
+      cap.type = `PanContinuous`;
+      cap.comment = getSpeedGuessedComment(capabilityName, cap);
+      return cap;
+    }
+
+    Object.assign(cap, getPanTiltCap(`Pan`, panMax));
+    cap.comment = capabilityName;
+
+    return cap;
+  },
+  Tilt: ({ channelName, capabilityName, tiltMax }) => {
+    const cap = {};
+
+    if (channelName.match(/continuous/i)) {
+      cap.type = `TiltContinuous`;
+      cap.comment = getSpeedGuessedComment(capabilityName, cap);
+      return cap;
+    }
+
+    Object.assign(cap, getPanTiltCap(`Tilt`, tiltMax));
+    cap.comment = capabilityName;
+
+    return cap;
+  },
+  Prism: ({ capabilityName }) => ({
+    type: `Prism`,
+    comment: capabilityName
+  }),
+  Shutter: ({ capabilityName }) => {
+    const cap = {
+      type: `ShutterStrobe`
+    };
+
+    if (capabilityName.match(/^(?:Blackout|(?:Shutter |Strobe )?Closed?)$/i)) {
+      cap.shutterEffect = `Closed`;
+      return cap;
+    }
+
+    if (capabilityName.match(/^(?:(?:Shutter |Strobe )?Open|Full?)$/i)) {
+      cap.shutterEffect = `Open`;
+      return cap;
+    }
+
+    if (capabilityName.match(/puls/i)) {
+      cap.shutterEffect = `Pulse`;
+    }
+    else if (capabilityName.match(/ramp\s*up/i)) {
+      cap.shutterEffect = `RampUp`;
+    }
+    else if (capabilityName.match(/ramp\s*down/i)) {
+      cap.shutterEffect = `RampDown`;
+    }
+    else {
+      cap.shutterEffect = `Strobe`;
+    }
+
+    if (capabilityName.match(/random/i)) {
+      cap.randomTiming = true;
+    }
+
+    cap.comment = getSpeedGuessedComment(capabilityName, cap);
+
+    return cap;
+  },
+  Speed: ({ channelName, capabilityName }) => {
+    const cap = {};
+
+    if (channelName.match(/pan(?:\/)?tilt/i)) {
+      cap.type = `PanTiltSpeed`;
+    }
+    else {
+      cap.type = `Speed`;
+    }
+
+    cap.comment = getSpeedGuessedComment(capabilityName, cap);
+
+    return cap;
+  }
+};
+
+/**
+ * Try to guess speedStart / speedEnd from the capability name and set them
+ * to the capability. It may also set cap.type to "Rotation".
+ * @param {string} capabilityName The capability name to extract information from.
+ * @param {object} cap The OFL capability object to add found properties to.
+ * @returns {string} The rest of the capabilityName.
+ */
+function getSpeedGuessedComment(capabilityName, cap) {
```

Comment:
This function sets some properties, but to ensure a consistent property order, a lot of extra logic is needed from where it is called. Furthermore, there are some (unexpected) side effect it does, for example setting the capability type.

It may be worth refactoring this to return an object `{ type, speedStart, speedEnd, comment }` where any of the properties may be an empty string. Then set the cap data like so:

```js
const { type, speedStart, speedEnd, comment } = getSpeedData(capabilityName, cap);
cap.type = type || cap.type;
cap.effectName = comment;
if (speedStart) {
  cap.shakeSpeedStart = speedStart;
  cap.shakeSpeedEnd = speedEnd;
}
```